lib/matplotlib/backends/_backend_tk.py

```diff
@@ -359,11 +360,20 @@ def flush_events(self):
     def start_event_loop(self, timeout=0):
         # docstring inherited
         if timeout > 0:
-            self._master.after(int(1000*timeout), self.stop_event_loop)
+            milliseconds = int(1000 * timeout)
+            if milliseconds > 0:
+                self._event_loop_id = self._tkcanvas.after(
+                    milliseconds, self.stop_event_loop)
+            else:
+                self._event_loop_id = self._tkcanvas.after_idle(
+                    self.stop_event_loop)
```

Comment:
If there are a lot of events, does `after_idle` have any particular timing? Since we know that `timeout > 0`, maybe we should just use `max(int(1000 * timeout), 1)` and clip the minimum?

Reply:
I chose `after_idle` mainly to be consistent with the behavior of `TkTimer`. In both cases, clipping the timeout to 1 ms would be a completely valid design choice. That said, an event loop with more small events in it than can be cleared in 1 ms could be considered pathological, and large events (like `draw_idle`) will cause unbounded delay regardless.

Reply:
OK, I guess that's fine if we're consistent about it.